denonavr/denonavr.py

```diff
@@ -871,6 +880,38 @@ def _get_support_sound_mode(self):
         else:
             return self._get_support_sound_mode_avr()
 
+    def _update_tone_control(self):
+        """Update tone control related things."""
+        try:
+            root = self.exec_appcommand_post(attribute_list=['GetToneControl'])
+        except requests.exceptions.ConnectTimeout:
+            root = None
+
+        if root is None:
+            _LOGGER.error("Getting tone control failed.")
+            return False
+
+        try:
+            self._tone_control_status = bool(int(root[0].find('status').text))
+        except (AttributeError, IndexError, TypeError):
+            return False
+
+        if self._tone_control_status is False:
+            # Tone Control can not be activated.
+            # e.g.: Due to active DynamicEQ
```

Comment:
Wouldn't it make sense to reset the tone control parameters in this case?
```python
self._tone_control_adjust = None
self._bass = None
self._bass_level = None
self._treble = None
self._treble_level = None
```
Activating Dynamic EQ via remote control or web interface deactivates tone control. If the values are not reset, we won't recognize that.

Reply:
Good catch 👍  Maybe I have sometime in the future to also add Dynamic EQ, so that we can also manage that 😄 
First I will try to bring this tone control feature somehow to Home Assistant 😃